test/test_datapipe.py

```diff
@@ -2067,43 +2083,127 @@ def test_traverse_mixdatapipe(self):
 class TestCircularSerialization(TestCase):
 
     class CustomIterDataPipe(IterDataPipe):
-        def add_one(self, x):
+
+        @staticmethod
+        def add_one(x):
```

Comment:
We still need a function attached to this instance because the function might needs some info from `self` like:
```py
class CustomIterDataPipe(IterDataPipe):
    def __init__(self, ...):
        self.v = 1
    def add_v(self, x):
        return x + self.v
```

Reply:
This is the corner case we want to cover.

Reply:
Do you intend to pass `add_v` to say `self.dp.map(add_v)`? Or just what you typed out?

May be related:
https://github.com/pytorch/pytorch/pull/74984#discussion_r863292137

Reply:
In any case, I can add that to the test